packages/react-native-reanimated/src/hook/useAnimatedStyle.ts

```diff
@@ -150,7 +151,11 @@ function runAnimations(
         animation.callback && animation.callback(true /* finished */);
       }
     }
-    result[key] = animation.current;
+    if (typeof animation.current === 'object') {
+      result[key] = { ...animation.current };
+    } else {
+      result[key] = animation.current;
+    }
```

Comment:
Why do we need this?

Reply:
because if we don't do that then not a copy is passed to result, rather a reference. It cause the color to not behave properly as it is being change in more than one place. Spreading the object fixes it

Reply:
Can you give a sample case? Why doesn't it affect other objects, only `boxShadow`?

Reply:
I'm not sure if we have other example than boxShadow itself. For example `backgroundColor` does simple value copying -> `result[key] = 'rgba(255,0,0,1)` but if we look at `boxShadow` to does it like that:
```
result[key] = { offsetX: 10, offsetY: 5, blurRadius: 3, color: 'rgba(233,0,0,1)
```
and by that it saves reference to object in `result[key]`

Reply:
I know we talked about this offline, but I don't remember the conclusion. `transform` cannot be animated like object as well? Does this affect how `transform` property animates?

Reply:
`transform` is fine, 'cause it only affects how `color` property is handled.

Normally we have `newValues` which in this case should hold not parsed `color` value. But if it was passed to `result[key]` (aka `updates[key]`) as an object, the `updates` held it as a reference resulting in both updates and newValues `color` prop being parsed. If both of them are parsed the animation cannot run, the `newValues` must stay in `rgba` format, to be assigned to `state.last`, to run the animation.

tl;dr it's all about `color` prop, every other prop is ok

Reply:
Then let's annotate that with a comment here.